src/Network/Request.php

```diff
@@ -158,6 +159,13 @@ class Request implements ArrayAccess
     protected $_session;
 
     /**
+     * Store the additional attributes attached to the request.
+     *
+     * @var array
+     */
+    protected $attributes = [];
```

Comment:
I want to move away from `_` in the long term. Public properties have proven to be really fragile for us and really awkward to refactor/work with.